client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,476 @@
+                let txs_to_resend_over_tpu = unconfirmed_transasction_map
+                    .iter()
+                    .filter(|x| blockheight < x.last_valid_blockheight)
+                    .map(|x| x.serialized_transaction.clone())
+                    .collect();
+                let _ = tpu_client
+                    .try_send_wire_transaction_batch(txs_to_resend_over_tpu)
+                    .await;
+
+                let elapsed = instant.elapsed();
+                if elapsed < TPU_RESEND_REFRESH_RATE {
+                    tokio::time::sleep(TPU_RESEND_REFRESH_RATE - elapsed).await;
+                }
+            } else {
+                tokio::time::sleep(Duration::from_millis(100)).await;
+            }
+            if let Some(max_valid_block_height_in_remaining_transaction) =
+                unconfirmed_transasction_map
+                    .iter()
+                    .map(|x| x.last_valid_blockheight)
+                    .max()
+            {
+                max_valid_block_height = max_valid_block_height_in_remaining_transaction;
+            }
+        }
+
+        if let Some(progress_bar) = progress_bar {
+            set_message_for_confirmed_transactions(
+                progress_bar,
+                num_confirmed_transactions.load(std::sync::atomic::Ordering::Relaxed),
+                total_transactions,
+                Some(current_blockheight.load(Ordering::Relaxed)),
+                max_valid_block_height,
+                "Checking transaction status...",
+            );
+        }
+    }
+}
+
+// This is a new method which will be able to send and confirm a large amount of transactions
+// usually the send_and_confirm_transactions_with_spinner for QUIC clients are optimized for small number of transactions
+// If we use these methods
+pub async fn send_and_confirm_transactions_in_parallel<T: Signers + ?Sized>(
+    rpc_client: Arc<RpcClient>,
+    tpu_client: Option<QuicTpuClient>,
+    messages: &[Message],
+    signers: &T,
+    config: SendAndConfrimConfig,
+) -> Result<Vec<Option<TransactionError>>> {
+    // get current blockhash and corresponding last valid block height
+    let (blockhash, last_valid_blockheight) = rpc_client
+        .get_latest_blockhash_with_commitment(rpc_client.commitment())
+        .await?;
+    let blockhash_data_rw = Arc::new(RwLock::new(BlockHashData {
+        blockhash,
+        last_valid_blockheight,
+    }));
+
+    // check if all the messages are signable by the signer
+    messages
+        .iter()
+        .map(|x| {
+            let mut transaction = Transaction::new_unsigned(x.clone());
+            transaction.try_sign(signers, blockhash)
+        })
+        .collect::<std::result::Result<Vec<()>, SignerError>>()?;
+
+    // get current blockheight
+    let block_height = rpc_client.get_block_height().await?;
+    let current_blockheight = Arc::new(AtomicU64::new(block_height));
```

Comment:
for consistency, I would use `block_height` everywhere because it looks like in the other parts of repo we stick with `block_height`

```suggestion
    let current_block_height = Arc::new(AtomicU64::new(block_height));
```